scripts/load_env.sh

```diff
@@ -17,65 +17,56 @@ set -o errexit
-  has_dupes=$(yq e '.. | select(. == "*") | {(path | .[-1]): .}| keys' config.yaml | sort| uniq -d)
-  if [ -n "${has_dupes:-}" ]; then
-    echo -e "\e[31m»»» 💥 There are duplicate keys in your config, please fix and try again!\e[0m"
-    exit 1
-  fi
+    fi
 
-  # Validate config schema
-  if [[ $(pajv validate -s config_schema.json -d config.yaml) != *valid* ]]; then
-    echo -e "\e[31m»»» ⚠️ Your config.yaml is invalid 😥 Please fix the errors and retry."
-    exit 1
-  fi
+    # Validate config schema
+    if [[ $(pajv validate -s "$schema_file_path" -d "$config_file_path") != *valid* ]]; then
+      echo -e "\e[31m»»» ⚠️ Your config is invalid 😥 Please fix the errors and retry."
+      exit 1
+    fi
 
-  # Get leaf keys yq query
-  GET_LEAF_KEYS=".. | select(. == \"*\") | {(path | .[-1]): .}"
-  # Map keys to uppercase yq query
-  UPCASE_KEYS="with_entries(.key |= upcase)"
-  # Suffix keys with TF_VAR_ yq query
-  TF_KEYS="with_entries(.key |= \"TF_VAR_\" + .)"
-  # Yq query to format the output to be in form: key=value
-  FORMAT_FOR_ENV_EXPORT="to_entries| map(.key + \"=\" +  .value)|join(\" \")"
+    # Get leaf keys yq query
+    GET_LEAF_KEYS=".. | select(. == \"*\") | {(path | .[-1]): .}"
+    # Map keys to uppercase yq query
+    UPCASE_KEYS="with_entries(.key |= upcase)"
+    # Suffix keys with TF_VAR_ yq query
+    TF_KEYS="with_entries(.key |= \"TF_VAR_\" + .)"
+    # Yq query to format the output to be in form: key=value
+    FORMAT_FOR_ENV_EXPORT="to_entries| map(.key + \"=\" +  .value)|join(\" \")"
 
-  # Export as UPPERCASE keys env vars
-  # shellcheck disable=SC2046
-  export $(yq e "$GET_LEAF_KEYS|$UPCASE_KEYS| $FORMAT_FOR_ENV_EXPORT" config.yaml)
-  # Export as Terraform keys env vars
-  # shellcheck disable=SC2046
-  export $(yq e "$GET_LEAF_KEYS|$TF_KEYS| $FORMAT_FOR_ENV_EXPORT" config.yaml)
+    # Export as UPPERCASE keys env vars
+    # shellcheck disable=SC2046
+    export $(yq e "$GET_LEAF_KEYS|$UPCASE_KEYS| $FORMAT_FOR_ENV_EXPORT" "$config_file_path")
+    # Export as Terraform keys env vars
+    # shellcheck disable=SC2046
+    export $(yq e "$GET_LEAF_KEYS|$TF_KEYS| $FORMAT_FOR_ENV_EXPORT" "$config_file_path")
 
-  # Get IP address for local deployments
-  if [[ "${LOCAL_MODE}" == "true" ]];
-  then
-      echo "Local Mode: TRUE"
-      if [[ -z "${DEPLOYER_IP_ADDRESS+x}" ]];
-      then
-          echo "No IP address assigned in config, getting client IP and setting in ENV"
-          this_ip="$(curl -s 'https://api64.ipify.org')"
-          export DEPLOYER_IP_ADDRESS="${this_ip}"
-      else
-          echo "Have IP address from config.yaml"
-      fi
-      echo "IP Address: ${DEPLOYER_IP_ADDRESS}"
-  else
-      echo "Local Mode: FALSE"
   fi
+}
 
-fi
-
+script_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
+# Export core config
+echo "Loading core configuration..."
+export_config_from_yaml "${script_dir}/../config.yaml" "${script_dir}/../config_schema.json"
```

Comment:
nit: I might do `if config_yaml_exists` here for readability

Reply:
Not sure I follow...?

Reply:
sorry, meant something like:

```bash
yaml_path="${script_dir}/../config.yaml"
if [ ! -f $yaml_path ]; then
    export_config_from_yaml $yaml_path
fi
```

cause it makes more sense to only export it if it exists but also meh, I'm not convinced in my suggestion either. bash is just unreadable